pyphs/numerics/simulations/simulation.py

```diff
@@ -353,39 +353,46 @@ def process(files):
         # close_files(files)
 
     def _process_cpp(self):
-
-        # build simu.cpp
+        
+                # build simu.cpp
         simu2cpp(self)
 
         # go to build folder
         os.chdir(self.cpp_path)
 
         # execute the bash script
-        self.system_call('./run.sh')
-
-        # Check if an error occured outside python (bash)
-        self._check_cpp_runtime_errors()
+        # self.system_call('./run.sh')
         
-        # Removes the error file if everything went well
-        os.remove('stderr')
+        # Commands
+        cmd = {
+               'Bbuild': '%s . -Bbuild' % self.config['cmake'],
+               'build': '%s --build build -- -j3' % self.config['cmake']
+              }
+        
+        # options for the subprocess.run method
+        sp_config = {
+            "cwd": self.cpp_path,
+            "stderr": subprocess.PIPE,
+            "check": True,
+            "universal_newlines": True,
+            "shell": True,
+            }
+        # Check compatibility with windows installs: 
+
+        # Perform build
+        with open(os.path.join(sp_config["cwd"], "build.log"), 'w') as fid:
+            sp_config['stdout'] = fid
+            p = subprocess.run(cmd['Bbuild'], **sp_config)
```

Comment:
You may wrap these two build steps to catch errors, and add a mention to the log in the build.log file. Something like:
```python

        with open(os.path.join(sp_config["cwd"], "build.log"), 'w') as fid:
            sp_config['stdout'] = fid
            try:
                p = subprocess.run(cmd['Bbuild'], **sp_config)
                p = subprocess.run(cmd['build'], **sp_config)
            except subprocess.CalledProcessError as err:
                print('Error while building simulation executable. See %s for details' % fid.name)
                print(p.stdout)
                print(p.stderr)
                raise subprocess.CalledProcessError(err)
```